src/sentry/interfaces/security.py

```diff
@@ -354,6 +354,18 @@ def from_raw(cls, raw):
 
         # For CSP, the values we want are nested under the 'csp-report' key.
         raw = raw['csp-report']
+
+        if 'effective-directive' not in raw:
+            # For Firefox, parse effective-directive from violated-directive
+            # https://bugzil.la/1192684#c8
+            violated_directive = raw['violated-directive'].split(None, 1)
+            if violated_directive:
+                raw['effective-directive'] = violated_directive[0]
+
+            # revalidate the raw data against the input schema (raises on failure)
```

Comment:
Why do we need to re-pass this through?

Reply:
This was to validate the parsed 'effective-directive' exists and matches the enum, but I see we're doing that in the interface schema so I dropped it.